Gentest failure: selected floats have different values in sqlite and in-memory db

Failure is [here](https://github.com/opensafely-core/databuilder/actions/runs/3945405128/jobs/6752267705).

```
FAILED tests/generative/test_query_model.py::test_query_model - AssertionError: Mismatch between in_memory and mssql
assert [{'patient_id...191345214844}] == [{'patient_id...019134521485}]
  At index 0 diff: {'patient_id': 1, 'v': 0.00037360191345214844} != {'patient_id': 1, 'v': 0.0003736019134521485}
  Full diff:
  - [{'patient_id': 1, 'v': 0.0003736019134521485}]
  ?                                             ^
  + [{'patient_id': 1, 'v': 0.00037360191345214844}]
  ?                                             ^^
Falsifying example: test_query_model(
    variable=SelectColumn(source=SelectPatientTable(name='p0', schema=TableSchema(i1=Column(int, constraints=()), i2=Column(int, constraints=()), b1=Column(bool, constraints=()), b2=Column(bool, constraints=()), d1=Column(datetime.date, constraints=()), d2=Column(datetime.date, constraints=()), f1=Column(float, constraints=()), f2=Column(float, constraints=()))), name='f2'),
    data=[{'type': data_setup.P0,
      'patient_id': 1,
      'i1': 0,
      'i2': 0,
      'b1': False,
      'b2': False,
      'd1': datetime.date(2000, 1, 1),
      'd2': datetime.date(2000, 1, 1),
      'f1': 0.0,
      'f2': 0.00037360191345214844}],
    query_engines={'in_memory': <tests.conftest.QueryEngineFixture at 0x7feb4b156970>,
     'sqlite': <tests.conftest.QueryEngineFixture at 0x7feb4b156a60>,
     'mssql': <tests.conftest.QueryEngineFixture at 0x7feb4b432790>},
    recorder=<tests.generative.recording.Recorder at 0x7feb4b439bb0>,
)

You can reproduce this example by temporarily adding @reproduce_failure('6.56.3', b'AXicY2ZgZGRkwA0O7AORrA9BJAAW2AJr') as a decorator on your test case
```

Presumably this is expected and we should be testing "equality" differently for floats.

--- a/tests/generative/test_query_model.py
+++ b/tests/generative/test_query_model.py
@@ -146,9 +146,21 @@ def run_test(query_engines, data, variable, recorder):
         if IGNORE_RESULT in [first_results, second_results]:
             continue  # pragma: no cover
 
-        assert (
-            first_results == second_results
-        ), f"Mismatch between {first_name} and {second_name}"
+        # If the results contain floats then we want only approximate equality to account
+        # for rounding differences
+        if any(
+            isinstance(v, float)
+            for res in [*first_results, *second_results]
+            for v in res.values()
+        ):
+            for i, result in enumerate(first_results):
+                assert result == pytest.approx(
+                    second_results[i], rel=1e-5
+                ), f"Mismatch between {first_name} and {second_name}"
+        else:
+            assert (
+                first_results == second_results
+            ), f"Mismatch between {first_name} and {second_name}"
 
 
 def run_error_test(query_engines, data, variable):

